exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/StringFunctions.java

```diff
@@ -416,16 +417,25 @@ public void setup() {
 
     @Override
     public void eval() {
-      if (index.value < 1) {
+      if (index.value == 0) {
         throw org.apache.drill.common.exceptions.UserException.functionError()
-          .message("Index in split_part must be positive, value provided was "
-            + index.value).build();
+          .message("Index in split_part can not be zero").build();
       }
       String inputString = org.apache.drill.exec.expr.fn.impl.
         StringFunctionHelpers.getStringFromVarCharHolder(in);
-      int arrayIndex = index.value - 1;
-      String result =
-              (String) com.google.common.collect.Iterables.get(splitter.split(inputString), arrayIndex, "");
+      String result = "";
+      if (index.value < 0) {
+        java.util.List<String> splits = splitter.splitToList(inputString);
+        int size = splits.size();
+        int arrayIndex = size + index.value;
+        if (arrayIndex >= 0) {
```

Comment:
What happens when arrayIndex < 0 here?  An informative error would be better than unitialised result data

Reply:
> What happens when arrayIndex < 0 here? An informative error would be better than uninitialised result data

arrayIndex < 0 is not an error, it will return empty string, see the case below in UT.
    testBuilder()
      .sqlQuery("select split_part('a,b,c', ',', -4) res1 from (values(1))")
      .ordered()
      .baselineColumns("res1")
      .baselineValues("")
      .go();